types/gapi/index.d.ts

```diff
@@ -264,7 +264,7 @@ declare namespace gapi.client {
      */
      class HttpRequestPromise<T> {
         // Taken and adapted from https://github.com/Microsoft/TypeScript/blob/v2.3.1/lib/lib.es5.d.ts#L1343
-        then<TResult1 = T, TResult2 = never>(
+        then<TResult1, T, TResult2, never>(
```

Comment:
This doesn't look right - now `then` has 4 type parameters. It has entirely different semantics than the previous revision.
I checked the header - when it says TypeScript Version 2.3 , it means the typings are intended to be used with TS 2.3 or higher. So I think the current state of the code is correct. 

What exactly are you trying to fix?